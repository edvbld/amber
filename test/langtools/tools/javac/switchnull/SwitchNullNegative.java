/**
 * @test /nodynamiccopyright/
 * @compile/fail/ref=SwitchNullNegative.out -XDrawDiagnostics SwitchNullNegative.java
 */

public class SwitchNullNegative {
    private int notReference(int i) {
        switch (i) {
            case null: return 0;
            case 1: return 1;
            default: return 2;
        }
    }

    private int notReferenceExpression(int i) {
        return switch (i) {
            case null -> 0;
            case 1 -> 1;
            default -> 2;
        };
    }

    private int repeatedStatement(Integer i) {
        return switch (i) {
            case 0: break 0;
            case null: break -1;
            case null: break -1;
            case 1: break 1;
        };
    }

    private int repeatedExpression(Integer i) {
        return switch (i) {
            case 0 -> 0;
            case null -> -1;
            case null -> -1;
            case 1 -> 1;
        };
    }
}
