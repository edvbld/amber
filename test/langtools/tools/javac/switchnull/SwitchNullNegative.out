SwitchNullNegative.java:9:13: compiler.err.switch.null.must.be.reference
SwitchNullNegative.java:17:13: compiler.err.switch.null.must.be.reference
SwitchNullNegative.java:27:13: compiler.err.duplicate.case.label
SwitchNullNegative.java:36:13: compiler.err.duplicate.case.label
4 errors
